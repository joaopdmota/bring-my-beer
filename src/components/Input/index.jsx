import React from 'react'
import styled, { css } from 'styled-components'
import { InputBase } from '../../utils/styles'

const InputStyled = styled.input`
  ${css`
    ${InputBase}
  `}
`

const InputWrapper = styled.div`
  flex-grow: 2;
  margin-bottom: 0;
  position: relative;
`

const Message = styled.p`
  top: 100%;
  margin: 4px 0px 0px;
  font-size: 11px;
  position: absolute;
  left: 2px;
  ${props =>
    props.error &&
    css`
      color: ${props => props.theme.error};
    `}
`

export default (props: Object) => {
  const { message = '', error = false } = props

  return (
    <InputWrapper>
      <InputStyled {...props} />
      <Message error={error} visible={message && message !== ''}>
        {message}
      </Message>
    </InputWrapper>
  )
}
