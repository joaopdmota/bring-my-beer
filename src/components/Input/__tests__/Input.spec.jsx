import React from 'react'
import Input from '../index'
import { shallow } from 'enzyme'

describe('Input Component', () => {
  it('render default Input', () => {
    const input = <Input />
    expect(input).toMatchSnapshot()
  })
  it('render default Input with message (not error)', () => {
    const message = 'Foo bar'
    const input = shallow(<Input message={message} />)
    expect(input.text().includes(message)).toBeTruthy()
    expect(input).toMatchSnapshot()
  })
  it('render default Input with error message', () => {
    const message = 'Foo bar'
    const input = shallow(<Input message={message} error />)
    expect(input.text().includes(message)).toBeTruthy()
    expect(input).toMatchSnapshot()
  })
})
