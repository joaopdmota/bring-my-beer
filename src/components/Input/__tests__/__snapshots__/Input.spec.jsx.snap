// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component render default Input 1`] = `<_default />`;

exports[`Input Component render default Input with error message 1`] = `
<Input__InputWrapper>
  <Input__InputStyled
    error={true}
    message="Foo bar"
  />
  <Input__Message
    error={true}
    visible={true}
  >
    Foo bar
  </Input__Message>
</Input__InputWrapper>
`;

exports[`Input Component render default Input with message (not error) 1`] = `
<Input__InputWrapper>
  <Input__InputStyled
    message="Foo bar"
  />
  <Input__Message
    error={false}
    visible={true}
  >
    Foo bar
  </Input__Message>
</Input__InputWrapper>
`;
